packages/app-store/_utils/writeAppDataToEventType.ts

```diff
@@ -0,0 +1,75 @@
+import prisma from "@calcom/prisma";
+import type { AppCategories } from "@calcom/prisma/enums";
+import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
+
+import type { appDataSchemas } from "../apps.schemas.generated";
+
+const writeAppDataToEventType = async ({
+  userId,
+  teamId,
+  appSlug,
+  appCategories,
+  credentialId,
+}: {
+  userId?: number;
+  teamId?: number;
+  appSlug: string;
+  appCategories: AppCategories[];
+  credentialId: number;
+}) => {
+  //   Search for event types belonging to the user / team
+  const eventTypes = await prisma.eventType.findMany({
```

Comment:
It could be a good candidate for a method in `EventTypeRepository`. The name could be `findByTeamOrUserOwner`.